lib/imagemounter/utils/list_developer_image.js

```diff
@@ -0,0 +1,174 @@
+ */
+async function findDeveloperImageFromDirectory(entry) {
+    const fileList = await readdir(entry);
+    for (const subEntry of fileList) {
+        const fullPath = joinPath(entry, subEntry);
+        const statResult = await lstat(fullPath);
+        if (subEntry === DEVELOPER_IMAGE_FILE_NAME && statResult.isFile()) {
+            return entry;
+        }
+        if (statResult.isDirectory()) {
+            const subFolderResult = await findDeveloperImageFromDirectory(fullPath);
+            if (subFolderResult) {
+                return subFolderResult;
+            }
+        }
+    }
+    return undefined;
+}
+
+/**
+ * @typedef {Object} ImagePath
+ * @property {string} developerImage
+ * @property {string} developerImageSignature
+ */
+
+/**
+ * Find developer image for certain version. If developer image does not exists,
+ * this will try to find and download developer image, unzip to `${APPIUM_HOME}/iOSSupport/`
+ * @param {string} version full version of iOS device. The first two parts of version
+ * will be preserved when sending the request, e.g. `14.7.1` will be changed to `14.7`
+ * @param {ImageFromGithubRepo} githubImageOption
+ * @returns {Promise<ImagePath>}
+ * @throws If developer image is not found, or error while downloading or unzipping.
+ */
+async function findDeveloperImage(version, githubImageOption) {
+    const finalVersion = version.split('.').splice(0, 2).join('.');
+    const fileName = `${finalVersion}.zip`;
+    const DEFAULT_IMAGE_DIR = joinPath(await env.resolveAppiumHome(), DEFAULT_IMAGE_DIR_NAME);
+    const fullDownloadPath = joinPath(DEFAULT_IMAGE_DIR, fileName);
+    if (!await exists(DEFAULT_IMAGE_DIR)) {
+        await mkdir(DEFAULT_IMAGE_DIR, { recursive: true });
+    }
+    if (!await exists(fullDownloadPath)) {
+        await searchAndDownloadDeveloperImageFromGithub(finalVersion, fullDownloadPath, githubImageOption);
+    }
+    const decompressPath = joinPath(DEFAULT_IMAGE_DIR, finalVersion);
+    let developerImageParentFolder = null;
+    if (await exists(decompressPath)) {
+        developerImageParentFolder = await findDeveloperImageFromDirectory(decompressPath);
+    }
+    if (!developerImageParentFolder) {
+        await zip.extractAllTo(fullDownloadPath, decompressPath);
+        developerImageParentFolder = await findDeveloperImageFromDirectory(decompressPath);
+    }
+    if (!developerImageParentFolder) {
+        throw new Error(`Unable to find unzipped developer image in ${decompressPath}`);
+    }
+    return {
+        developerImage: joinPath(developerImageParentFolder, DEVELOPER_IMAGE_FILE_NAME),
+        developerImageSignature: joinPath(developerImageParentFolder, DEVELOPER_IMAGE_SIGNATURE_FILE_NAME)
+    };
+}
+
+async function searchAndDownloadDeveloperImageFromGithub(finalVersion, fullDownloadPath, githubImageOption) {
+    const fileNameRegExp = new RegExp(`${finalVersion}(\\(([\\w_|.()])+\\))?.zip`);
```

Comment:
not sure if we also need to properly escape `finalVersion` string. Maybe use https://www.geeksforgeeks.org/lodash-_-escaperegexp-method/

Reply:
you are right, `.` should definitely be escaped